core/src/test/java/org/elasticsearch/index/seqno/GlobalCheckpointSyncIT.java

```diff
@@ -0,0 +1,218 @@
+
+                            senderTransportService.addDelegate(receiverTransportService,
+                                    new MockTransportService.DelegateTransport(senderTransportService.original()) {
+                                        @Override
+                                        protected void sendRequest(
+                                                final Connection connection,
+                                                final long requestId,
+                                                final String action,
+                                                final TransportRequest request,
+                                                final TransportRequestOptions options) throws IOException {
+                                            if ("indices:admin/seq_no/global_checkpoint_sync[r]".equals(action)) {
+                                                throw new IllegalStateException("blocking indices:admin/seq_no/global_checkpoint_sync[r]");
+                                            } else {
+                                                super.sendRequest(connection, requestId, action, request, options);
+                                            }
+                                        }
+                                    });
+                        }
+                    }
+                },
+                client -> {
+                    // restore global checkpoint syncs between all nodes
+                    final DiscoveryNodes nodes = client.admin().cluster().prepareState().get().getState().getNodes();
+                    for (final DiscoveryNode node : nodes) {
+                        for (final DiscoveryNode other : nodes) {
+                            if (node == other) {
+                                continue;
+                            }
+                            final MockTransportService senderTransportService =
+                                    (MockTransportService) internalCluster().getInstance(TransportService.class, node.getName());
+                            final MockTransportService receiverTransportService =
+                                    (MockTransportService) internalCluster().getInstance(TransportService.class, other.getName());
+                            senderTransportService.clearRule(receiverTransportService);
+                        }
+                    }
+                });
+    }
+
+    private void runGlobalCheckpointSyncTest(
+            final TimeValue globalCheckpointSyncInterval,
+            final Consumer<Client> beforeIndexing,
+            final Consumer<Client> afterIndexing) throws Exception {
+        final int numberOfReplicas = randomIntBetween(1, 4);
+        internalCluster().ensureAtLeastNumDataNodes(1 + numberOfReplicas);
+        // set the sync interval high so it does not execute during this test
```

Comment:
this comment belongs to the calling method.

Reply:
I pushed b80b728d968887b91b7be101d137fc71b44e523f.